pkg/ensurance/collector/nodelocal/disk.go

```diff
@@ -0,0 +1,123 @@
+package nodelocal
+
+import (
+	"io/ioutil"
+	"time"
+
+	"github.com/shirou/gopsutil/disk"
+	"k8s.io/klog/v2"
+
+	"github.com/gocrane/crane/pkg/common"
+	"github.com/gocrane/crane/pkg/ensurance/collector/types"
+)
+
+const (
+	diskioCollectorName = "diskio"
+	sysBlockPath        = "/sys/block"
+)
+
+func init() {
+	registerMetrics(diskioCollectorName, []types.MetricName{types.MetricDiskReadKiBPS, types.MetricDiskWriteKiBPS, types.MetricDiskReadIOPS, types.MetricDiskWriteIOPS, types.MetricDiskUtilization}, NewDiskIOCollector)
+}
+
+type DiskTimeStampState struct {
+	stat      disk.IOCountersStat
+	timestamp time.Time
+}
+
+type DiskIOCollector struct {
+	diskStates map[string]DiskTimeStampState
+}
+
+type DiskIOUsage struct {
+	DiskReadKiBps  float64
+	DiskWriteKiBps float64
+	DiskReadIOps   float64
+	DiskWriteIOps  float64
+	Utilization    float64
+}
+
+// NewDiskIOCollector returns a new Collector exposing kernel/system statistics.
+func NewDiskIOCollector(_ *NodeLocalContext) (nodeLocalCollector, error) {
+
+	klog.V(2).Infof("NewDiskIOCollector")
+
+	return &DiskIOCollector{diskStates: make(map[string]DiskTimeStampState)}, nil
+}
+
+func (d *DiskIOCollector) collect() (map[string][]common.TimeSeries, error) {
+	var now = time.Now()
+
+	devices, err := SysBlockDevices(sysBlockPath)
+	if err != nil {
+		return map[string][]common.TimeSeries{}, err
+	}
+
+	diskIOStats, err := disk.IOCounters(devices...)
+	if err != nil {
+		klog.Errorf("Failed to collect disk io resource: %v", err)
+		return map[string][]common.TimeSeries{}, err
```

Comment:
suggest to return nil instead of map[string][]common.TimeSeries{} on error.

Reply:
done~